### ISSUE TITLE
CookieTempDataProvider should not call DeleteCookie just because TempData currently has no values

### ISSUE THREAD

**Issue by DanielStout5** [dotnet/aspnetcore#46189]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

On every request our app is checking TempData (using the default cookie storage) for several values.

On almost all requests, these values are not set - and yet, on every single response this header is attached:

`.AspNetCore.Mvc.CookieTempDataProvider=; expires=Thu, 01 Jan 1970 00:00:00 GMT; path=/; samesite=lax; httponly`

From what I can tell, that's because in `CookieTempDataProvider.SaveTempData`, if there are no TempData values, it calls `_chunkingCookieManager.DeleteCookie(context, _options.Cookie.Name, cookieOptions);`

The problem is that this will disable Output Caching since that only applies to requests that don't set cookies.

### Expected Behavior

Reading TempData without modifying it should not set a cookie.

### Steps To Reproduce

Just check TempData without modifying it. The "delete" cookie gets appended with every request.

https://github.com/DanielStout5/TempDataDeleteCookieExample

### .NET Version

7.0.101

### Anything else?

ASP.NET Core 7.0.2
Visual Studio 2022

**Comment by Eilon** [comment]
Yeah I think that logically "if there's was no TempData at the start of the request, and there's no TempData at the end of the request", then we should set no cookies at all.
